src/Raven.Server/Documents/Handlers/Processors/Indexes/IndexHandlerProcessorForConvertAutoIndex.cs

```diff
@@ -0,0 +1,716 @@
+                {
+                    case FieldStorage.Yes:
+                        options.Storage = FieldStorage.Yes;
+                        break;
+                    default:
+                        throw new ArgumentOutOfRangeException();
+                }
+            }
+
+            void HandleSuggestions(string fieldName, bool? suggestions)
+            {
+                if (suggestions.HasValue == false || suggestions == false)
+                    return;
+
+                var options = GetFieldOptions(fieldName);
+                options.Suggestions = true;
+            }
+
+            void HandleSpatial(string fieldName, AutoSpatialOptions spatialOptions, AutoIndexConversionContext context)
+            {
+                if (spatialOptions == null)
+                    return;
+
+                var realFieldName = context.FieldNameMapping[fieldName];
+
+                var options = GetFieldOptions(realFieldName);
+                options.Spatial = spatialOptions;
+            }
+        }
+    }
+
+    public string ConvertToAbstractIndexCreationTask(AutoIndexDefinition autoIndex, out string csharpClassName)
+    {
+        if (autoIndex == null)
+            throw new ArgumentNullException(nameof(autoIndex));
+
+        var context = new AutoIndexConversionContext();
+
+        var sb = new StringBuilder();
+
+        csharpClassName = GenerateClassName(autoIndex);
+        var className = autoIndex.Collection is Constants.Documents.Collections.AllDocumentsCollection or Constants.Documents.Collections.EmptyCollection
+            ? "object"
+            : Inflector.Singularize(autoIndex.Collection);
+
+        sb
+            .Append($"public class {csharpClassName} : {typeof(AbstractIndexCreationTask).FullName}<{className}");
+
+        if (autoIndex.Type == IndexType.AutoMapReduce)
+            sb.Append($", {csharpClassName}.Result");
+
+        sb
+            .AppendLine(">")
+            .AppendLine("{")
+            .AppendLine($"public {csharpClassName}()")
+            .AppendLine("{");
+
+        ConstructMap(autoIndex, sb, context);
+        ConstructReduce(autoIndex, sb);
+        ConstructFieldOptions(autoIndex, sb, context);
+
+        sb
+            .AppendLine("}");
+
+        if (autoIndex.Type == IndexType.AutoMapReduce)
+        {
+            sb
+                .AppendLine()
+                .AppendLine("public class Result")
+                .AppendLine("{");
+
+            foreach (var kvp in autoIndex.MapFields)
+            {
+                var fieldNames = GenerateFieldName(kvp.Key, kvp.Value.Indexing);
+
+                foreach (var f in fieldNames)
+                {
+                    switch (kvp.Value.Aggregation)
+                    {
+                        case AggregationOperation.Sum:
+                        case AggregationOperation.Count:
+                            {
+                                sb.AppendLine($"public int {f.FieldName} {{ get; set; }}");
```

Comment:
What about sum of floating point numbers? And would a single pair of brackets be enough?

Reply:
I do not know anything about the type, so wild guess here. User will have to make the correction if needed.